crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -593,11 +597,8 @@ impl SettlementEncoder {
             clearing_prices,
             trades,
             interactions: [
-                // In the following it is assumed that all different interactions
-                // are only required once to be executed.
```

Comment:
While writing a unit test for encoding the interactions I stumbled over this assumption. It didn't seem necessary to me so I dropped it but if it's actually required I can adjust my test instead.
Could this be a workaround required for multiple Ethflow orders in a settlement?
cc @fedgiac

Reply:
It is a workaround: it's going to work without but it's going to be gas inefficient if there are multiple eth-flow orders in the same settlement.
Eth-flow orders need a call to [`wrapAll`](https://etherscan.io/address/0xD02De8Da0B71E1B59489794F423FaBBa2AdC4d93#code) to wrap ETH. However, it's enough to call it once after all traded orders have been created.

Reply:
In that case it might be worthwhile to collapse all `wrapAll` interactions from the EthFlow contract into one and allow duplicates for anything else.
On the other hand there are probably only very few settlements which have multiple EthFlow orders where it would make a difference.

Reply:
This is an optimization that applies to EthFlow orders only. In the general sense, pre-interactions aren't "collapsible", so IMO we shouldn't do this in the general case.

Maybe, since this is technically a regression, can we create a GitHub issue to track it? I would prefer changing this code to handle the "general" pre-interaction case.

I agree that there are likely few settlements where this will make a difference, so I don't think this would make a meaningful difference to gas costs. @MartinquaXD - maybe worth asking the Data team during the "office hours" today, how many settlements have more than one EthFlow order settled at the same time. This should give us an exact Ξ value of gas money saved since EthFlow launch.